scripts/olean_comparison.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+ : << 'BASH_MODULE_DOCS'
+
+# Change in size of the `.olean`s
+
+This is the main script for the `compare oleans` CI step.
+It should take place after `lake build` has taken place, since it computes sizes of the oleans.
+
+There are two CI steps involved:
+* `print the sizes of the oleans` that simply tallies the sizes of all the folders where the
+  `.olean`s are stored;
+* `compare oleans` that essentially uses the script below to find differences in size between
+  the `.olean`s on master and the ones in the current PR.
+
+## The script
+
+First, the script retrieves the log of the CI step `print the sizes of the oleans` from the
+latest master run.
+These are the "reference" `.olean`s against which the differences are computed.
+
+Next, the script first prints the sizes of the oleans of the *current* branch.
+This means that on a PR you see the sizes of the newly created oleans
+and on master you see the sizes of the "comparison" oleans.
+
+Finally, the script runs the comparison between the "`master`" sizes and the sizes
+of the newly created oleans.
+It flags every folder that has a percentage change of at least 5% (either positive or negative).
+The percentage difference for the full `.olean`s folder is always printed, whether or not it
+exceeds the threshold.
+
+## Check
+
+When bors merges a batch into master, running this code show the net effect of the new code over
+the old one on the `.olean`s.
+Do the logs for the corresponding commit to `master` reproduce this data?
+
+BASH_MODULE_DOCS
+
+# Because the `lean-pr-testing-NNNN` branches use toolchains that are "updated in place"
+# the cache mechanism is unreliable, so we don't test it if we are on such a branch.
+if [[ $(cat lean-toolchain) =~ ^leanprover/lean4-pr-releases:pr-release-[0-9]+$ ]]; then
+  printf 'No summary, since this is a pr testing branch\n'
+fi
+
+oleansDir=.lake/build/lib/Mathlib
+
+# should be master
+branch=adomani/CI_olean_size
```

Comment:
```suggestion
branch=master
```
This should be changed before merging: it is the branch that is used as a comparison branch *and* it should contain the CI action that logged the sizes of the `olean`s.